### PR TITLE
[Data Exfiltration Detection] Update and add anomaly detection jobs and security rules

### PR DESCRIPTION
**Pull request by sodhikirti07**
## What does this PR do?
- Adds filter to map source IP and destination IP to only contain internal and external addresses respectively.
- Changes the configurations of current anomaly detection jobs.
- Changed the description of detectors in ml module
- Adds two anomaly detection jobs and rules to identify exfiltration to external devices (USB, Airdrop)
- Updates the lookback time of the detection rules i.e. 2*bucket_span of ml job.
- Updates README, changelog, manifest and dashboard.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

## Screenshots

Find them in the comment section

### CODE REVIEW

**Comment by sodhikirti07** [comment]
### Screenshots

- Overview page

**Comment by susan-shu-c** [comment]
I feel that I'd avoid creating a lot of false positives for new hosts 🤔 re: `host.name`

2nd question, in the description's second point, `Changes the configurations of current anomaly detection jobs.`, what are the main changes?

Lastly, is it possible to explain how the Airdrop was tested?

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 1]
```diff
@@ -161,7 +168,8 @@
                             "user.name",
                             "process.name",
                             "source.ip",
-                            "destination.ip"
+                            "destination.ip",
+                            "destination.geo.city_name"
```
Comment (peteharverson) [inline packages/ded/kibana/ml_module/ded-ml.json:172]: What was the reason for going with `geo.city_name` as an extra influencer, but not using `region_name`?

**Comment by sodhikirti07** [comment]
> 2nd question, in the description's second point, Changes the configurations of current anomaly detection jobs., what are the main changes?

@susan-shu-c 
I updated the filter query to include internal source and external destination IPs. The anomaly detection jobs are now partitioned by `host.name` , earlier we were doing population analysis. Lastly, deleted some redundant jobs like anomalies looking at city names, continent names and added them as influencers instead (for info gain). See the current jobs [here](https://github.com/elastic/integrations/blob/main/packages/ded/kibana/ml_module/ded-ml.json)

> Lastly, is it possible to explain how the Airdrop was tested?

Tested by setting up a macOS VM using Parallels desktop and also confirmed with Ricardo Ungureanu.

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 2]
```diff
@@ -30,21 +30,22 @@
...
                         "data_exfiltration"
                     ],
-                    "description": "A machine learning job to detect data exfiltration to an unusual geo-location (by city name).",
+                    "description": "Detects data exfiltration to an unusual geo-location (by country iso code).",
                     "analysis_config": {
                         "bucket_span": "3h",
                         "detectors": [
                             {
-                                "detector_description": "high_sum(\"source.bytes\") over \"destination.geo.city_name\"",
+                                "detector_description": "high_sum(\"source.bytes\") over \"destination.geo.country_iso_code\"",
```
Comment (peteharverson) [inline packages/ded/kibana/ml_module/ded-ml.json:44]: All the detectors in the jobs are using the default descriptions which are not always the most user-friendly. These are used in various places in the UI - for example in the ML anomalies table. Any thoughts on going with something which is closer to the form of text used in the job descriptions? For example `High bytes sent to an unusual country_iso_code`? I know we often go with the defaults, but just wondering if there is something we could do here?

**Review by peteharverson — COMMENTED** [review]
Overall looks like some great updates here. Left a couple of comments on the detector descriptions and the influencers for one of the ML jobs.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 1 above)
Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json:172]: Added for additional context only, `region_name` is already there as a field.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 2 above)
Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json:44]: TBH, I didn't try to touch them as we always followed the default for these jobs. I'm down trying simple text as a descriptor, but IDK, If it's okay to deviate one package's structure from the rest.

File: packages/ded/docs/README.md [hunk 3]
```diff
@@ -10,37 +10,37 @@ Then use these detection rules and anomaly detection jobs for data exfiltration
...
+| Rule | Description                                                                                                    |
+|---|----------------------------------------------------------------------------------------------------------------|
+| Potential Data Exfiltration Activity to an Unusual ISO Code | An anomaly detection job to detect an abnormal volume of bytes being sent to an unusual country by its iso code. |
+| Potential Data Exfiltration Activity to an Unusual Region | An anomaly detection job to detect an abnormal volume of bytes being sent to an unusual region name.           |
+| Potential Data Exfiltration Activity to an Unusual IP Address | An anomaly detection job to detect an abnormal volume of bytes being sent to an unusual IP address.            |
+| Potential Data Exfiltration Activity to an Unusual Destination Port | An anomaly detection job to detect an abnormal volume of bytes being sent to an unusual destination port.      |
+ | High bytes of data sent to an external device | An anomaly detection job to detect high bytes of data written to an external device.                           |
```
Comment (ajosh0504) [inline packages/ded/docs/README.md:41]: Please change rule names to Title Case.

File: packages/ded/docs/README.md [hunk 4]
```diff
@@ -10,37 +10,37 @@ Then use these detection rules and anomaly detection jobs for data exfiltration
...
+| High bytes of data sent to an external device via Airdrop | An anomaly detection job to detect high bytes of data written to an external device via Airdrop. |
```
Comment (ajosh0504) [inline packages/ded/docs/README.md:42]: Please change rule names to Title Case.

File: packages/ded/docs/README.md [hunk 5]
```diff
@@ -10,37 +10,37 @@ Then use these detection rules and anomaly detection jobs for data exfiltration
...
+ | Rare process writing data to an external device | An anomaly detection job to detect a rare process writing data to an external device.                          |
```
Comment (ajosh0504) [inline packages/ded/docs/README.md:43]: Please change rule names to Title Case.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 2 above)
Comment (ajosh0504) [reply packages/ded/kibana/ml_module/ded-ml.json:44]: I'd agree with Pete. We've switched to what he's suggesting for our pre-built ML rules. I'd say let's change this package to begin with, and also update others in later versions.

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 6]
```diff
@@ -138,21 +144,22 @@
                 }
             },
             {
-                "id": "high-sent-bytes-destination-geo-country_name",
+                "id": "high-sent-bytes-destination-region_name",
```
Comment (ajosh0504) [inline packages/ded/kibana/ml_module/ded-ml.json:147]: Any reason for retaining the region name job, in addition to ISO code?

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 7]
```diff
@@ -210,30 +219,31 @@
...
             {
-                "id": "high-sent-bytes-destination-port",
+                "id": "rare-process-writing-to-external-device",
                 "config": {
                     "groups": [
                         "security",
                         "data_exfiltration"
                     ],
-                    "description": "A machine learning job to detect data exfiltration to an unusual destination port.",
+                    "description": "Detects data exfiltration activity by identifying a writing event started by a rare process to an external device.",
```
Comment (ajosh0504) [inline packages/ded/kibana/ml_module/ded-ml.json:228]:
```suggestion
                    "description": "Detects data exfiltration activity by identifying a file write started by a rare process to an external device.",
```

File: packages/ded/kibana/security_rule/798c1290-6411-486e-bc85-7ae9fe826aaf.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        "anomaly_threshold": 75,
+        "author": [
+            "Elastic"
+        ],
+        "description": "An anomaly detection job to detect high bytes of data written to an external device.\n",
+        "from": "now-2h",
+        "interval": "15m",
+        "license": "Elastic License",
+        "machine_learning_job_id": "high-bytes-written-to-external-device",
+        "name": "High bytes of data sent to an external device",
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/798c1290-6411-486e-bc85-7ae9fe826aaf.json:12]:
```suggestion
        "name": "Spike in Bytes Sent to an External Device",
```

File: packages/ded/kibana/security_rule/7a342a6b-58df-4832-919d-d4bd681e3138.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        "anomaly_threshold": 75,
+        "author": [
+            "Elastic"
+        ],
+        "description": "An anomaly detection job to detect a rare process writing data to an external device.\n",
+        "from": "now-2h",
+        "interval": "15m",
+        "license": "Elastic License",
+        "machine_learning_job_id": "rare-process-writing-to-external-device",
+        "name": "Rare process writing data to an external device",
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/7a342a6b-58df-4832-919d-d4bd681e3138.json:12]:
```suggestion
        "name": "Unusual Process Writing Data to an External Device",
```

File: packages/ded/kibana/security_rule/da5cd97e-7f3c-40dc-9183-7ef1270e8817.json [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        "anomaly_threshold": 75,
+        "author": [
+            "Elastic"
+        ],
+        "description": "An anomaly detection job to detect high bytes of data written to an external device via Airdrop.\n",
+        "from": "now-2h",
+        "interval": "15m",
+        "license": "Elastic License",
+        "machine_learning_job_id": "high-bytes-written-to-external-device-airdrop",
+        "name": "High bytes of data sent to an external device via Airdrop",
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/da5cd97e-7f3c-40dc-9183-7ef1270e8817.json:12]:
```suggestion
        "name": "Spike in Bytes Sent to an External Device via Airdrop",
```

**Review by ajosh0504 — APPROVED** [review]
Love the new detections! Made some minor comments.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 6 above)
Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json:147]: They both have different set of granularities, so thought would be nice to see anomalies at both levels.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 2 above)
Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json]: Changed the description of detectors

image

**Review by susan-shu-c — APPROVED** [review]

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 1 above)
Comment (peteharverson) [reply packages/ded/kibana/ml_module/ded-ml.json:172]: But `destination.geo.region_name` is not included in the `influencers` for this job. The `over` field name would normally be included as an influencer.

**Review by peteharverson — APPROVED** [review]
Changes for the ML jobs LGTM

**Comment by sodhikirti07** [comment]
Made the following changes based on my discussion with Dian:
- Changed the ml-module query to work with either file or network events.
- Changed the anomaly detection jobs to their original state, i.e., removed partition by `host.name`. The `host.name` field is not compatible with all the data sources for network logs. Identifiers such as `source.ip` are the most applicable here but creates a hard limit on the model due to their high cardinalities.

Haven't changed the partition by field in the exfiltration to external device jobs. Let me know the consensus of folks in using `user.name` or `host.name` as the partitioning field.

cc: @ajosh0504 @susan-shu-c @dainperkins

**Comment by susan-shu-c** [comment]
To document our discussion, I'll defer to Dain for now on using `user`, though agree that there are places it's not so clear-cut.

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 11]
```diff
@@ -2,19 +2,27 @@
...
-                "filter": [
+                "should": [
                     {
                         "term": {
-                            "event.category": "network"
+                            "event.category": "file"
                         }
                     },
+                    {
+                        "bool":{
```
Comment (ajosh0504) [inline packages/ded/kibana/ml_module/ded-ml.json:19]: Do you need this bool? Also do you have to check for existence of source bytes and destination fields? Are they not present in all network events?

File: packages/ded/kibana/security_rule/5576098d-650d-4b0f-a3a1-48b5b0ce4f15.json [hunk 12]
```diff
@@ -20,7 +20,7 @@
             "ML"
         ],
         "type": "machine_learning",
-        "version": 3
+        "version": 5
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/5576098d-650d-4b0f-a3a1-48b5b0ce4f15.json:23]: Should be 4?

File: packages/ded/kibana/security_rule/8d4e2e5f-982b-4e8d-b765-a10d648efc33.json [hunk 13]
```diff
@@ -20,7 +20,7 @@
             "ML"
         ],
         "type": "machine_learning",
-        "version": 3
+        "version": 5
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/8d4e2e5f-982b-4e8d-b765-a10d648efc33.json:23]: Should be 4?

File: packages/ded/kibana/security_rule/b9d01e71-ac5d-4203-82ed-c1246d6c72c8.json [hunk 14]
```diff
@@ -20,7 +20,7 @@
             "ML"
         ],
         "type": "machine_learning",
-        "version": 3
+        "version": 5
```
Comment (ajosh0504) [inline packages/ded/kibana/security_rule/b9d01e71-ac5d-4203-82ed-c1246d6c72c8.json:23]: Should be 4?

File: packages/ded/kibana/security_rule/5576098d-650d-4b0f-a3a1-48b5b0ce4f15.json (hunk 12 above)
Comment (sodhikirti07) [reply packages/ded/kibana/security_rule/5576098d-650d-4b0f-a3a1-48b5b0ce4f15.json:23]: Ah! yes, thanks for catching.

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 15]
```diff
@@ -2,19 +2,27 @@
...
             "bool": {
-                "filter": [
+                "should": [
                     {
                         "term": {
-                            "event.category": "network"
+                            "event.category": "file"
                         }
                     },
+                    {
```
[the remark this answers is not in the document]
Comment (sodhikirti07) [reply packages/ded/kibana/ml_module/ded-ml.json:18]: Done!

**Comment by sodhikirti07** [comment]
Based on my discussion with @susan-shu-c and @ajosh0504, I am keeping the current `partition_by` configurations in place i.e., partitioning the jobs by `host.name`.

**Comment by dainperkins** [comment]
2 things - 
1) USB exfil - from a security and behavior modeling standpoint,  in any environment with shared workstations this will work much better if partitioned by user, as hosts will invariably be less aligned to a specific behavior profile, and it seemed like most file events will have a user involved

2) if the partitioning by host.name for network exfil is used the query needs to only pull data with host.name (which I think is just packetbeat) otherwise the model is going to be somewhat useless for any network data that isn't provided by packetbeat (e.g. netflow, vpc flow, firewwall logs, sensors, etc.)  

Either the partitioning needs to be removed or set to source.ip, or the detection needs to be specific to data that includes host.name.  Otherwise the partitioning will be effectively based on packet beat data by host.name, and everything else (including non packetbeat data for the same hosts if any other network data is being ingested. 

from an enterprise standpoint the idea of deploying packetbeat / network capture wideliy, instead of sensors or netflow, is resource prohibitive, and nothing I would ever want to implement in the real world.

**Comment by dainperkins** [comment]
Also if Packetbeat is used as a sensor (e.g. spanning a network tap) every doc will have the same host.name (the host name of the device receiving the network span)

**Comment by sodhikirti07** [comment]
> Either the partitioning needs to be removed or set to source.ip, or the detection needs to be specific to data that includes host.name.

@dainperkins the partitionings on network exfil jobs are already removed. 

> USB exfil - from a security and behavior modeling standpoint, in any environment with shared workstations this will work much better if partitioned by user, as hosts will invariably be less aligned to a specific behavior profile, and it seemed like most file events will have a user involved

From an ML standpoint, we think it's better to use `host.name` as a partitioning field for USB based jobs. Also, the influencers section adds the user related info to the alerts and anomalies.

**Comment by dainperkins** [comment]
sorry - thought the following was in regards to **all** the jobs:

> Based on my discussion with @susan-shu-c and @ajosh0504, I am keeping the current partition_by configurations in place i.e., partitioning the jobs by host.name.

**Pull request merged**